### PR TITLE
[Merged by Bors] - chore(Analysis/NormedSpace/QuaternionExponential): extract helper lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
Besides being split in three, this proof is largely untouched.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/QuaternionExponential.lean [hunk 1]
```diff
@@ -35,6 +35,48 @@ theorem exp_coe (r : ℝ) : exp ℝ (r : ℍ[ℝ]) = ↑(exp ℝ r) :=
   (map_exp ℝ (algebraMap ℝ ℍ[ℝ]) (continuous_algebraMap _ _) _).symm
 #align quaternion.exp_coe Quaternion.exp_coe
 
+/-- The even terms of the series are real, and correspond to the series for `cos`. -/
+theorem expSeries_even_of_imaginary {q : Quaternion ℝ} (hq : q.re = 0) (n : ℕ):
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/QuaternionExponential.lean:39]:
```suggestion
/-- The even terms of `expSeries` are real, and correspond to the series for `cos`. -/
theorem expSeries_even_of_imaginary {q : Quaternion ℝ} (hq : q.re = 0) (n : ℕ) :
```

File: Mathlib/Analysis/NormedSpace/QuaternionExponential.lean [hunk 2]
```diff
@@ -35,6 +35,48 @@ theorem exp_coe (r : ℝ) : exp ℝ (r : ℍ[ℝ]) = ↑(exp ℝ r) :=
...
+    _ = ↑((-1 : ℝ) ^ n * ‖q‖ ^ (2 * n) / k) := ?_
+  · congr 1
+    rw [neg_pow, normSq_eq_norm_mul_self, pow_mul, sq]
+    push_cast
+    rfl
+  · rw [← coe_mul_eq_smul, div_eq_mul_inv]
+    norm_cast
+    ring_nf
+
+/-- The even terms of the series are real, and correspond to the series for
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/QuaternionExponential.lean:57]:
```suggestion
/-- The odd terms of `expSeries` are real, and correspond to the series for
```

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Analysis/NormedSpace/QuaternionExponential.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/QuaternionExponential.lean:57]: Whoops

File: Mathlib/Analysis/NormedSpace/QuaternionExponential.lean [hunk 3]
```diff
@@ -35,6 +35,48 @@ theorem exp_coe (r : ℝ) : exp ℝ (r : ℍ[ℝ]) = ↑(exp ℝ r) :=
   (map_exp ℝ (algebraMap ℝ ℍ[ℝ]) (continuous_algebraMap _ _) _).symm
 #align quaternion.exp_coe Quaternion.exp_coe
 
+/-- The even terms of `expSeries` are real, and correspond to the series for `cos`. -/
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/QuaternionExponential.lean:38]:
```suggestion
/-- The even terms of `expSeries` are real, and correspond to the series for $\cos ‖q‖$. -/
```

File: Mathlib/Analysis/NormedSpace/QuaternionExponential.lean [hunk 4]
```diff
@@ -35,6 +35,48 @@ theorem exp_coe (r : ℝ) : exp ℝ (r : ℍ[ℝ]) = ↑(exp ℝ r) :=
...
+  · congr 1
+    rw [neg_pow, normSq_eq_norm_mul_self, pow_mul, sq]
+    push_cast
+    rfl
+  · rw [← coe_mul_eq_smul, div_eq_mul_inv]
+    norm_cast
+    ring_nf
+
+/-- The odd terms of `expSeries` are real, and correspond to the series for
+$\frac{\sin ‖q‖}{‖q‖} q$. -/
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/QuaternionExponential.lean:58]:
```suggestion
$\frac{q}{‖q‖} \sin ‖q‖$. -/
```

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**